### ISSUE TITLE
Passing class level annotations

### ISSUE THREAD

**Issue by solid-tarka** [immutables/immutables#116]
What are the implications of passing class level annotations?

    @JsonSerialize(as = TestRepresentation.class) 
    @JsonDeserialize(as = TestRepresentation.class) 
    @Value.Immutable 
    @RepresentationStyle 
    @SomeCustomAnnotation("some_custom_annotation_property") 
    public interface Test { 
    ... 
    } 

Translated to: 

    @SuppressWarnings("all") 
    @Generated({"Immutables.generator", "Test"}) 
    @SomeCustomAnnotation("some_custom_annotation_property") 
    public final class TestRepresentation implements Test { 
    ... 
    } 

My actual use case is with a Jackson JSON-LD library. The library converts an object (ideally immutable) into a JSON linked data representation. The class level annotation is critical as it defines the @context of the object.

    @JsonldType("http://schema.org/Person")
    public class Person { 
        @JsonldProperty("http://schema.org/name")
        public String name;
    }

This gets serialised by jackson to:

    {
        "@context": {
            "name": "http://schema.org/name",
        },
        "name": "John Doe",
    }

**Comment by elucash** [comment]
Thank you for the feature request. The most difficult choice is how to control (or not) which annotations are getting copied.

**Comment by solid-tarka** [comment]
I agree. 
I'm not sure how other people are using Immutables.org but from my perspective I only really create either interfaces or abstract. For me at least it would be possible to actually wrap annotations that I wanted to pass.

So instead of this which is difficult to control:
```
@Value.Immutable 
@SomeCustomAnnotation("some_custom_annotation_property") 
public interface Test { 
... 
} 
```
We create an annotation wrapper like this:

```
@Value.Immutable 
@Value.Annotations(value = "@SomeCustomAnnotation(\"some_property\")") 
public interface Test { 
... 
} 
```

and the outcome is:
```
@SuppressWarnings("all") 
@Generated({"Immutables.generator", "Test"}) 
@SomeCustomAnnotation("some_custom_annotation_property") 
public final class TestRepresentation implements Test { 
... 
} 
```

**Comment by elucash** [comment]
@augustotravillio as you currently messing around annotations for serialization, could you check if you can get along with this issue? Need to be able to pass some safe subset of annotations

**Comment by augustotravillio** [comment]
Looking into it. most safe way would be to add a white list of annotations to the style. Styles would allow to apply to a range of value objects. Passing all annotations could cause problems, and passing annotation as a string would be verbose and error prone (no type safety, no IDE help).

**Comment by elucash** [comment]
sounds like a plan, call it `passAnnotations` or something along the lines.

**Comment by elucash** [comment]
Style attribute `passAnnotations` was added in released version 2.0.12.
You can play with it to see if it fits your needs. Styles could be applied on types, packages or as meta-annotations, use this to share common white-list of annotations to copy over.
